src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -327,7 +402,31 @@ cdef class _AioCall(GrpcCallWrapper):
         metadata_sent_observer()
 
         # Receives initial metadata.
-        initial_metadata_observer(
-            await _receive_initial_metadata(self,
-                                            self._loop),
+        self._set_initial_metadata(
+            await _receive_initial_metadata(self, self._loop)
         )
+
+
+cdef _AioCall new_AioCall(AioChannel channel, object deadline,
```

Comment:
@lidizheng If we follow CPython C API naming conventions, this would be named `PyAioCall_New`.

Reply:
Happy to change the factory to something that we all agree. In principle, I would prefer to have the static method rather than have a function with a lot of similarities with the ones coming from the CPython API, basically for avoiding confusion when the CPP file is read.

But before moving forward with this discussion lets try to close this one first [1], where @gnossen challenges - If I'm not wrong - the idea of not using the traditional `__ciinit__`.

[1] https://github.com/grpc/grpc/pull/21696#discussion_r368107403

Reply:
Closing this conversations since we will be using the traditional `__cini__` method